backend/src/main/scala/cromwell/backend/wdl/WriteFunctions.scala

```diff
@@ -27,8 +32,21 @@ trait WriteFunctions extends IoFunctionSet with AsyncIoFunctions {
     }
   }
 
+  private val relativeToLocal = System.getProperty("user.dir")
+  private val relativeToPapi = ConfigFactory.load().as[Option[String]]("papi.default-input-gcs-prefix").orElse(ConfigFactory.load().as[Option[String]]("backend.providers.JES.config.root")).getOrElse("gs://")
+
+  private def withSlash(str: String) = if (str.endsWith("/")) str else str + "/"
+
+  def relativeToAbsolutePath(pathFrom: String, papi: Boolean): String = if (new File(pathFrom).isAbsolute) pathFrom else {
+    val result = withSlash(if (papi) relativeToPapi else relativeToLocal) + pathFrom
+    println(s"RelativeToAbsolute of $pathFrom was $result")
+    result
+  }
+
   override def copyFile(pathFrom: String, targetName: String): Future[WomSingleFile] = {
-    val source = _writeDirectory.root / pathFrom
+    val papi = buildPath(".").toString.startsWith("gs")
```

Comment:
ToL: I will be _strongly_ encouraging each cloud vendor to also contribute tests with their code. OSS tests are still incoming...

Otherwise we'll have no idea if/when problems arise. Eventually if we don't allow the teams to pass in a method variable, eventually multiple methods like this one will contain local variables committed like:

```scala
val papi = buildPath(".").toString.startsWith("gs")
val oss = buildPath(".").toString.startsWith("oss")
val s3 = buildPath(".").toString.startsWith("s3")
val azureBlob = buildPath(".").toString.contains(".blob.core.windows.net")
…
```

Reply:
Is this only to get the conformance tests working or is it more than that ?
If it's just for the conformance test I wish there was a less disruptive way to do it :(

Reply:
afaict these inline variables are "get papi-conformance tests working". I _hope_ we eventually see PRs for "get tes-conformance tests working", "get azure-conformance tests working", "get bcs-conformance tests working", "get aws-conformance tests working", etc. otherwise nobody knows if cromwell still works with those backends. 🤷‍♀️

Reply:
totally agree, I was just saying that I would really like to see if there's really no other way than ending up with stuff like 

```
val papi = buildPath(".").toString.startsWith("gs")
val oss = buildPath(".").toString.startsWith("oss")
val s3 = buildPath(".").toString.startsWith("s3")
val azureBlob = buildPath(".").toString.contains(".blob.core.windows.net")
```

all over the place

Reply:
Also, to be clearer, these contributor-specific additions will continue to be added to this generic `cromwell.backend.wdl` namespace.

@mcovarr's use of similar jes-conformance-config-injection are isolated within `cromwell.backend.impl.jes`.

https://github.com/broadinstitute/cromwell/blob/1a429769544021c4ddb45a81e1096f52a0988c55/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesParameters.scala#L17-L22

Reply:
FWIW this comes up because the conformance tests aren't meant for cloud platforms and server mode. If we did the fully-qualifying in the conformance runner (for JES, not for local) rather than crowbarring it here I'd be so much happier.

Reply:
What about adding a:

```scala
package wom.expression
trait IoFunctionSet {
  def toAbsolute(path: String): String
```

Backend contributors already need to extend IoFunctionSet, and could add any absolute-path in an override of this method.